### DISCUSSION TITLE
Where are the sources of the lua classes?

### DISCUSSION THREAD

**Discussion by CodeSandwich** [jgm/pandoc#9508] [Q&A] [answered]
The documentation of the lua classes doesn't cover all the edge cases, which is probably fine for a FOSS project where the source code is available that can answer all the weird questions. But where exactly is the lua classes source code? It doesn't seem to be in this repository, is it generated from the Haskell code?

**Comment by jgm** [comment, marked as answer]
A lot of these are simply wrapped versions of Haskell functions.
Some of the code is in pandoc-lua-engine.  Other parts are in separate packages that pandoc-lua-engine imports.
If you're wondering about something specific, we  might be able to give more informative answers.
